### PR TITLE
CLI tests

### PR DESCRIPTION
**Pull request by mckerracher**
I didn't update the makefile because the updates would just be commented out anyways. If you want to run any of these tests, just add them to the makefile really quick.

### CODE REVIEW

File: test/test12.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+export LD_LIBRARY_PATH=build/
+if build/marc --debug output=tmp.bin input/test2.opl; then
```
Comment (kedar-damle) [inline test/test12.sh:4]: This will fail .. did you mean --output?

File: test/test13.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+export LD_LIBRARY_PATH=build/
+if build/marc output=tmp.bin input/test2.opl; then
```
Comment (kedar-damle) [inline test/test13.sh:4]: Same as test12

File: test/test10.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+export LD_LIBRARY_PATH=build/
+if build/marc --debug output=test.bin; then
```
Comment (kedar-damle) [inline test/test10.sh:4]: Are you testing lack of required argument? In that case the other argument should be --output. Else this fails but not as per the test.

File: test/test11.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+export LD_LIBRARY_PATH=build/
+if build/marc --gobbletygook output=test.bin; then
```
Comment (kedar-damle) [inline test/test11.sh:4]: Here the test seems to be for invalid argument and inaccessible input file?. Can you split these two tests? If that is not the intention, provide a valid input file.

File: test/test14.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+export LD_LIBRARY_PATH=build/
+if build/marc --debug output=tmp.bin input/test2.opl  StarTrek>StarWars; then
```
Comment (kedar-damle) [inline test/test14.sh:4]: Similar to test11, inaccessibl input file, too many arguments test together.

File: test/test15.sh [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+export LD_LIBRARY_PATH=build/
+if build/marc output=tmp.bin input/test2.opl  StarTrek>StarWars; then
```
Comment (kedar-damle) [inline test/test15.sh:4]: Same as 14

**Review by kedar-damle — CHANGES_REQUESTED** [review]
Hey Josh,

1. Can you add a short description of each test to the tests-desc.md file?

2. In a test, can you run the program by itself and check the exit code reported? Right now, the test will just look for exit code being 0 or not. We also want to test the individual error codes returned when the program fails. For example in this case, program fails because it interprets output=tmp.bin as the source file.

```
$ export LD_LIBRARY_PATH=build/
$ build/marc --debug output=tmp.bin input/test2.opl
access('output=tmp.bin', F_OK): No such file or directory
$ echo $?
2
```
As you can see the exit code returned corresponds to ENOENT.
```
$ errno 2
ENOENT 2 No such file or directory
```
You can test it like this
```
#!/bin/bash
export LD_LIBRARY_PATH=build/
build/marc --debug output=tmp.bin input/test2.opl
if [[ $? -ne 2 ]] ; then
  exit 1
fi
exit 0
```

File: test/test12.sh (hunk 1 above)
Comment (mckerracher) [reply test/test12.sh]: I did mean --output. Thanks for catching that!

File: test/test13.sh (hunk 2 above)
Comment (mckerracher) [reply test/test13.sh]: fixed

**Review by kedar-damle — APPROVED** [review]

**Pull request merged**